src/RPC/middleware.ts

```diff
@@ -8,6 +8,7 @@ import type {
 import { TransformStream } from 'stream/web';
 import * as rpcErrors from './errors';
 import * as rpcUtils from './utils';
+import { promise } from '../utils';
 const jsonStreamParsers = require('@streamparser/json');
```

Comment:
Do not use `require`. Use `import` or `import()`. This should be deprecated. We only use it in very specific set of cases.

Reply:
Import is acting weird with this library. when I use the import semantics or `import()` I get the following error.  `TS2307: Cannot find module '@streamparser/json' or its corresponding type declarations.`

I think it's using module exports. We have problems importing that right?

Reply:
You can't use `import()` without top level await or being in Async context.

Reply:
But you should ask upstream and check what you can do

Reply:
It's not an upstream issue. If I asked they'd likely just point to the first line of their example `import { JSONParser } from '@streamparser/json';`. It's more likely a problem with our configuration, I'll look into it some more.

Reply:
Why can't we import like that?

Reply:
Because when I do I get `TS2307: Cannot find module '@streamparser/json' or its corresponding type declarations.`.

Reply:
We don't have this problem with any of the other libraries. Although I think if they are using the new node module system instead of commonjs, then it can be an issue. But we should be working out a way to be able to import both module systems and not be using `require`.

Reply:
I think you should indicate this upstream. That way you can point out something about our configuration causing import problems, but that doesn't affect other packages we have been using.

Reply:
Upstream was quite responsive to our previous problems too involving DoS.

Reply:
I'll make  a new issue and an upstream issue for this. It isn't blocking the PR.

Reply:
I have created a new issue for this at #516 .